crates/trigger/src/runtime_config.rs

```diff
@@ -132,34 +144,38 @@ impl RuntimeConfig {
 
 #[derive(Debug, Default, Deserialize)]
 #[serde(deny_unknown_fields)]
-struct RuntimeConfigOpts {
+pub struct RuntimeConfigOpts {
     #[serde(default)]
     pub state_dir: Option<String>,
 
     #[serde(default)]
     pub log_dir: Option<PathBuf>,
 
     #[serde(rename = "config_provider", default)]
-    pub config_providers: Vec<ConfigProvider>,
-}
+    pub config_providers: Vec<ConfigProviderOpts>,
 
-#[derive(Debug, Deserialize)]
-#[serde(rename_all = "snake_case", tag = "type")]
-pub enum ConfigProvider {
-    Vault(VaultConfig),
-}
+    #[serde(rename = "key_value_store", default)]
+    pub key_value_stores: HashMap<String, KeyValueStoreOpts>,
 
-// Vault config to initialize vault provider
-#[derive(Debug, Default, Deserialize)]
-#[serde(deny_unknown_fields)]
-pub struct VaultConfig {
-    pub url: String,
-    pub token: String,
-    pub mount: String,
-    pub prefix: Option<String>,
+    #[serde(skip)]
+    pub file_path: Option<PathBuf>,
 }
 
-type BoxedConfigProvider = Box<dyn spin_config::Provider>;
+fn resolve_config_path(path: &Path, config_opts: &RuntimeConfigOpts) -> Result<PathBuf> {
+    if path.is_absolute() {
+        return Ok(path.to_owned());
+    }
+    let base_path = match &config_opts.file_path {
+        Some(file_path) => file_path
+            .parent()
+            .with_context(|| {
+                format!("failed to get parent of runtime config file path {file_path:?}")
+            })?
+            .to_owned(),
+        None => std::env::current_dir().context("failed to get current directory")?,
```

Comment:
Nit: this error message just repeats the underlying error. We should provide context about runtime config like we do in the lines above.

Reply:
> this error message just repeats the underlying error

Not sure what you mean here; afaict `current_dir` just returns an `io::Error` without any calling context?